### PR TITLE
IO SBML Adds export of rxn/met note field

### PR DESCRIPTION
**Pull request by ChristianLieven**
Fixed: Notes field of reactions was not imported into cobrapy or cameo model as reaction.notes. It is now included as a dictionary, which makes parsing/ mapping easier.

Fixed: Notes field of reactions was not included in legacy sbml export from cobrapy or cameo model.

Fixed: When the formula of a metabolite is simply included as a string it will not be exported in the notes section correctly since the original code looked for the ID of a formula object. I changed this to check for a formula attribute regardless of its type. This should allow both options to be exported correctly.

### CODE REVIEW

**Comment by ChristianLieven** [comment]
Should be fixed now!

File: cobra/io/sbml.py [hunk 1]
```diff
@@ -291,6 +291,7 @@ def create_cobra_model_from_sbml_file(sbml_filename, old_sbml=False, legacy_meta
         if 'SUBSYSTEM' in reaction_note_dict:
             reaction.subsystem = reaction_note_dict['SUBSYSTEM'][0]   
 
+        reaction.notes = reaction_note_dict
```
Comment (aebrahim) [inline cobra/io/sbml.py]: We actually don't want to change the model when writing it out.

File: cobra/io/sbml.py [hunk 2]
```diff
@@ -490,13 +491,28 @@ def get_libsbml_document(cobra_model,
...
-                                                                 note_end_tag,
-                                                                 note_start_tag,
-                                                                 the_reaction.subsystem,
-                                                                 note_end_tag))
 
+        #Checks if GPR and Subsystem annotations are present in the notes section and if they are the same as those in
+        #the reaction's gene_reaction_rule/ subsystem attribute
+        #If they are not identical, they are set to be identical
+        if the_reaction.gene_reaction_rule:
+            the_reaction.notes.update({'GENE ASSOCIATION': [str(the_reaction.gene_reaction_rule)]})
```
Comment (aebrahim) [inline cobra/io/sbml.py]: Should be modifying reaction_note_dict, not the_reaction.notes

**Comment by aebrahim** [comment]
Also, if you do ```from six import iteritems``` you can do ```iteritems(some_dict)``` which works correctly in both python 2 and python 3.

**Comment by ChristianLieven** [comment]
Alright! I'll take care of that!

**Comment by aebrahim** [comment]
Replaced by #205. Please test that and make sure it works for you.

**Pull request closed without merging**